cotk/dataloader/dataloader.py

```diff
@@ -128,21 +141,79 @@ def _general_load_data(self, file_path, data_keys, \
 			* **data** (dict): a dict contains data.
 			* **data_size** (dict): a dict contains size of each item in data.
 		'''
+		if set(type_ for _, type_ in data_format) > {'label', 'sentence', 'session'}:
+			raise ValueError
+
+		def read_sentence(lines):
+			return next(lines).rstrip()
+
+		def read_session(lines):
+			session = []
+			while True:
+				try:
+					line = next(lines)
+					if line == '\n':
+						break
+					session.append(line.rstrip())
+				except StopIteration:
+					break
+			if not session:
+				raise StopIteration
+			return session
+
+		def read_label(lines):
+			label = int(next(lines).strip())
+			return label
+
+		read_functions = {
+			'label': read_label,
+			'sentence': read_sentence,
+			'session': read_session,
+		}
+
+		def read(lines, type_):
+			return read_functions[type_.lower().strip()](lines)
```

Comment:
You have checked the type_ must be in {label, sentence, session}, why do we need checks again?